util/src/journaldb.rs

```diff
@@ -170,7 +200,10 @@ impl JournalDB {
 			if backing.get(&h.bytes()).expect("Low-level database error. Some issue with your hard disk?").is_some() {
 				// already in the backing DB. start counting, and remember it was already in.
 				Self::set_already_in(batch, &h);
-				counters.insert(h.clone(), 1);
+				refs.insert(h.clone(), RefInfo{queue_refs: 1, in_archive: true});
+				if trace {
```

Comment:
`trace!` macro has this check internally. There is no performance penalty when tracing is disabled